pkg/reconciler/managed/reconciler.go

```diff
@@ -675,35 +894,55 @@ func (r *Reconciler) Reconcile(ctx context.Context, req reconcile.Request) (reco
 		"external-name", meta.GetExternalName(managed),
 	)
 
-	// Check the pause annotation and return if it has the value "true"
-	// after logging, publishing an event and updating the SYNC status condition
-	if meta.IsPaused(managed) {
-		log.Debug("Reconciliation is paused via the pause annotation", "annotation", meta.AnnotationKeyReconciliationPaused, "value", "true")
-		record.Event(managed, event.Normal(reasonReconciliationPaused, "Reconciliation is paused via the pause annotation"))
+	managementPoliciesEnabled := r.features.Enabled(feature.EnableAlphaManagementPolicies)
+	if managementPoliciesEnabled {
+		log.WithValues("managementPolicies", managed.GetManagementPolicies())
+	}
+
+	// Create the management policy resolver which will assist us in determining
+	// what actions to take on the managed resource based on the management
+	// and deletion policies.
+	policyResolver := NewManagementPoliciesResolver(managementPoliciesEnabled, managed.GetManagementPolicies(), managed.GetDeletionPolicy())
```

Comment:
```suggestion
	policy := NewManagementPoliciesResolver(managementPoliciesEnabled, managed.GetManagementPolicies(), managed.GetDeletionPolicy())
```

Naming nit: I'd prefer this because `policy.ShouldDelete()` reads more like "the policy is 'should delete'" to me. Not something I feel very strong about though.